numpy/core/tests/test_regression.py

```diff
@@ -1518,19 +1519,17 @@ def test_ticket_1539(self):
         dtypes = [x for x in np.typeDict.values()
                   if (issubclass(x, np.number)
                       and not issubclass(x, np.timedelta64))]
-        a = np.array([], dtypes[0])
+        a = np.array([], np.bool_)  # not x[0] because it is unordered
         failures = []
-        # ignore complex warnings
-        with warnings.catch_warnings():
-            warnings.simplefilter('ignore', np.ComplexWarning)
```

Comment:
The complex warning only happened on Python 3 because the dtypes dict is unsorted, while on python 2 `a` would always be `np.bool_` and none of the casts need the complex warning then. The cast is not being tested here though.